From 1ee3c3478ded17a2d8e6ac50901b60ed3a694ee0 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Thu, 4 Jan 2024 14:35:08 +0530
Subject: [PATCH 761/955] config/arm: add armv9-a march

Now that major versions of GCC recognize armv9-a march option,
add it to the list of supported march.
Update neoverse-n2 part number to include march as armv9-a.

Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Change-Id: I7c431fa70d4b0898dcbed2a870049bce7df623e0
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/119145
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
---
 config/arm/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/arm/meson.build b/config/arm/meson.build
index 6442ec95960b5..69b281ad6cacc 100644
--- a/config/arm/meson.build
+++ b/config/arm/meson.build
@@ -86,7 +86,7 @@ part_number_config_arm = {
         ]
     },
     '0xd49': {
-        'march': 'armv8.5-a',
+        'march': 'armv9-a',
         'march_features': ['sve2'],
         'flags': [
             ['RTE_MACHINE', '"neoverse-n2"'],
@@ -613,7 +613,7 @@ if update_flags
     # probe supported archs and their features
     candidate_march = ''
     if part_number_config.has_key('march')
-        supported_marchs = ['armv8.6-a', 'armv8.5-a', 'armv8.4-a', 'armv8.3-a',
+        supported_marchs = ['armv9-a', 'armv8.6-a', 'armv8.5-a', 'armv8.4-a', 'armv8.3-a',
                             'armv8.2-a', 'armv8.1-a', 'armv8-a']
         check_compiler_support = false
         foreach supported_march: supported_marchs
-- 
2.25.1

